packages/keyring-eth-onekey/src/onekey-keyring.ts

```diff
@@ -0,0 +1,717 @@
+};
+
+export type AccountPage = AccountPageEntry[];
+
+export type OneKeyControllerOptions = {
+  hdPath?: string;
+  accounts?: Hex[];
+  accountDetails?: Readonly<Record<string, AccountDetails>>;
+  page?: number;
+  passphraseState?: string;
+};
+
+export type OneKeyControllerState = {
+  hdPath: string;
+  accounts: string[];
+  accountDetails: Record<string, AccountDetails>;
+  page: number;
+  passphraseState?: string;
+};
+
+/**
+ * Check if the given transaction is made with ethereumjs-tx or @ethereumjs/tx
+ *
+ * Transactions built with older versions of ethereumjs-tx have a
+ * getChainId method that newer versions do not.
+ * Older versions are mutable
+ * while newer versions default to being immutable.
+ * Expected shape and type
+ * of data for v, r and s differ (Buffer (old) vs BN (new)).
+ *
+ * @param tx - Transaction to check, instance of either ethereumjs-tx or @ethereumjs/tx.
+ * @returns Returns `true` if tx is an old-style ethereumjs-tx transaction.
+ */
+function isOldStyleEthereumjsTx(
+  tx: TypedTransaction | OldEthJsTransaction,
+): tx is OldEthJsTransaction {
+  return 'getChainId' in tx && typeof tx.getChainId === 'function';
+}
+
+/**
+ * Check if the given value has a hex prefix.
+ *
+ * @param value - The value to check.
+ * @returns Returns `true` if the value has a hex prefix.
+ */
+function hasHexPrefix(value: string): boolean {
+  return value.startsWith('0x');
+}
+
+/**
+ * Add a hex prefix to the given value.
+ *
+ * @param value - The value to add a hex prefix to.
+ * @returns Returns the value with a hex prefix.
+ */
+function addHexPrefix(value: string): string {
+  if (hasHexPrefix(value)) {
+    return value;
+  }
+  return `0x${value}`;
+}
+
+/**
+ * Check if the passphrase state is empty.
+ *
+ * @param passphraseState - The passphrase state to check.
+ * @returns Returns `true` if the passphrase state is empty.
+ */
+function isEmptyPassphrase(passphraseState: string | undefined): boolean {
+  return (
+    passphraseState === null ||
+    passphraseState === undefined ||
+    passphraseState === ''
+  );
+}
+
+export class OneKeyKeyring implements Keyring {
+  readonly type: string = keyringType;
+
+  static type: string = keyringType;
+
+  page = 0;
+
+  perPage = 5;
+
+  unlockedAccount = 0;
+
+  hdk = new HDKey();
+
+  accounts: readonly Hex[] = [];
+
+  accountDetails: Record<string, AccountDetails> = {};
+
+  passphraseState: string | undefined;
+
+  hdPath = defaultHdPath;
+
+  network: NetworkApiUrls = NetworkApiUrls.Mainnet;
+
+  implementFullBIP44 = false;
+
+  bridge: OneKeyBridge;
+
+  eventEmitter: EventEmitter;
+
+  constructor({ bridge }: { bridge: OneKeyBridge }) {
+    if (!bridge) {
+      throw new Error('Bridge is a required dependency for the keyring');
+    }
+
+    this.eventEmitter = new EventEmitter();
+
+    this.bridge = bridge;
+    this.bridge.on(ONEKEY_HARDWARE_UI_EVENT, (_event: any) => {
+      this.eventEmitter.emit(ONEKEY_HARDWARE_UI_EVENT, _event);
+    });
```

Comment:
What is this event emitter used for? we tried to move away from using unsafe event emitters - any chance we can do something like what's being done in `eth-qr-keyring`?